src/memray/_memray/records.h

```diff
@@ -36,6 +36,14 @@ enum class RecordType {
     CONTEXT_SWITCH = 12,
 };
 
+struct RecordTypeAndFlags
```

Comment:
Can we have an assert somewhere that this will be able to hold the `RecordType` enum if we add new record types in the future? This can be a nightmare if someone just adds the record that doesn't fit :(

Reply:
Sure, but the `assert` will only trigger in debug builds, and the warning shows up in all builds - repeatedly, since it shows up once per file that includes `records.h`, which turns out to be most of them.

But, ok, adding the assert is easy enough.